source/m3_compile.c

```diff
@@ -584,7 +584,12 @@ M3Result  PushConst  (IM3Compilation o, u64 i_word, u8 i_type)
         {
             if (IsSlotAllocated (o, slot) and IsSlotAllocated (o, slot + 1))
             {
+# if d_m3Use32BitSlots
+                u64 constant;
+                memcpy (&constant, &o->constants [slot - o->slotFirstConstIndex], sizeof (u64));      // Avoid unaligned read
```

Comment:
I haven't measured the performance to verify this, but I think it is safe to keep this as a single code path for both 32-bit and 64-bit slots. The compiler should be able to replace `memcpy(dest, src, sizeof(uint64_t))` when `dest` and `src` are both pointers to `uint64_t`.
The performance may get slightly worse when building with -Og but at higher optimization levels this shouldn't make any difference, and the code will be simpler.